spyder/plugins/completion/kite/client.py

```diff
@@ -68,6 +68,16 @@ def get_languages(self):
             response = ['python']
         return response
 
+    def get_status(self, filename, saved=True):
+        """Get Kite status taking into account a filename."""
+        if saved:
+            verb, url = KITE_ENDPOINTS.SAVED_STATUS_ENDPOINT
+            url = url.format(filename=filename)
+        else:
+            verb, url = KITE_ENDPOINTS.UNSAVED_STATUS_ENDPOINT
```

Comment:
If we have a filename, we should always use the `filename=` URL.

The `filetype=` URL is for cases where the user is editing a python buffer but there is no filename (and there is no corresponding file on disk).

The side effect of getting this wrong is that there is a lot of status bar noise claiming that Kite has no index for the file, once the file goes into an unsaved state, even though that is not true.

Sorry if this wasn't clear in the spec.